### PR TITLE
Rework ima keys

### PR DESCRIPTION
**Pull request by trini**

### CODE REVIEW

File: meta-integrity/recipes-core/initrdscripts/files/init.ima [hunk 1]
```diff
@@ -95,20 +95,23 @@ fi
...
-keyring_id=0x`grep '\skeyring\s*\.ima: ' "${ROOT_DIR}/proc/keys" | awk '{ print $1 }'`
+mount --move ${ROOT_DIR}/proc /proc
 
 # The trusted IMA certificate /etc/keys/x509_evm.der in initramfs was
 # automatically loaded by kernel already. Here is the opportunity to load
 # a custom IMA certificate from the real rootfs.
 for cert in ${ROOT_DIR}/etc/keys/x509_evm*.der; do
     [ ! -s "$cert" ] && continue
+    name=`basename $cert`
+    echo $name
```
Comment (jiazhang0) [inline meta-integrity/recipes-core/initrdscripts/files/init.ima]: Remove this echo.

File: meta-integrity/recipes-core/initrdscripts/files/init.ima [hunk 2]
```diff
@@ -95,20 +95,23 @@ fi
...
 
-    if ! evmctl import "$cert" "$keyring_id" >"${ROOT_DIR}/dev/null"; then
+    if ! keyctl padd asymmetric "$name" %:.ima < $cert > ${ROOT_DIR}/dev/null; then
         print_critical "Unable to load the custom IMA certificate $cert for IMA appraisal"
```
Comment (jiazhang0) [inline meta-integrity/recipes-core/initrdscripts/files/init.ima]: keyutils needs to be added to RDEPENDS_${PN} of initrdscripts-ima.bb.

File: meta-integrity/recipes-core/initrdscripts/files/init.ima [hunk 3]
```diff
@@ -97,6 +97,20 @@ fi
 
 mount --move ${ROOT_DIR}/proc /proc
 
+# If we have a secondary trusted keyring, here is the opportunity to load
+# additional trusted keys from the real rootfs.
+for cert in ${ROOT_DIR}/etc/keys/x509_secondary_*.der; do
+    [ ! -s "$cert" ] && continue
+    name=`basename $cert`
+    echo $name
+
```
Comment (jiazhang0) [inline meta-integrity/recipes-core/initrdscripts/files/init.ima]: Please remove this echo.

File: meta-integrity/recipes-core/initrdscripts/files/init.ima (hunk 1 above)
Comment (trini) [reply meta-integrity/recipes-core/initrdscripts/files/init.ima]: Oops, debugging leftover.

File: meta-integrity/recipes-core/initrdscripts/files/init.ima (hunk 2 above)
Comment (trini) [reply meta-integrity/recipes-core/initrdscripts/files/init.ima]: I'll update the list, it's implicit today because ima-evm-utils needs libkeyutils

**Comment by jiazhang0** [comment]
@trini Nice work. Thanks!

**Pull request merged**